File: video-service-client/src/main/java/com/boclips/videos/service/client/spring/VideoServiceClientConfig.java
package com.boclips.videos.service.client.spring;

import com.boclips.videos.service.client.ServiceCredentials;
import com.boclips.videos.service.client.VideoServiceClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class VideoServiceClientConfig {

    @Value("${video-service.base-url}")
    public String baseUrl = null;

    @Value("${video-service.access-token-uri}")
    public String accessTokenUri = null;

    @Value("${video-service.client-id}")
    public String clientId = null;

    @Value("${video-service.client-secret}")
    public String clientSecret = null;

    @Bean
    public VideoServiceClient videoServiceClient() {
        if (baseUrl == null) {
            throw new IllegalStateException("Missing base url for video service client. Please configure video-service.base-url");
        }

        if (accessTokenUri == null) {
            throw new IllegalStateException("Missing base url for video service client. Please configure video-service.base-url");
        }

        if (clientId == null) {
            throw new IllegalStateException("Missing clientId for video service client. Please configure video-service.client-id");
        }

        if (clientSecret == null) {
            throw new IllegalStateException("Missing clientSecret for video service client. Please configure video-service.client-secret");
        }


        return VideoServiceClient.getApiClient(
                baseUrl,
                ServiceCredentials.builder()
                        .accessTokenUri(accessTokenUri)
                        .clientId(clientId)
                        .clientSecret(clientSecret).build()
        );
    }
}

File: video-service/src/client/kotlin/com/boclips/videos/service/client/spring/VideoServiceClientConfig.kt
package com.boclips.videos.service.client.spring

import com.boclips.videos.service.client.ServiceCredentials
import com.boclips.videos.service.client.VideoServiceClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
internal class VideoServiceClientConfig {

    @Value("\${videoService.baseUrl}")
    var baseUrl: String? = null

    @Value("\${videoService.accessTokenUri}")
    var accessTokenUri: String? = null

    @Value("\${videoService.clientId}")
    var clientId: String? = null

    @Value("\${videoService.clientSecret}")
    var clientSecret: String? = null

    @Bean
    fun videoServiceClient(): VideoServiceClient {
        return VideoServiceClient.getApiClient(
                baseUrl ?: throw IllegalStateException("Missing base url for video service client. Please configure video-service.base-url"),
                ServiceCredentials(
                        accessTokenUri = accessTokenUri ?: throw IllegalStateException("Missing base url for video service client. Please configure video-service.base-url"),
                        clientId = clientId ?: throw IllegalStateException("Missing clientId for video service client. Please configure video-service.client-id"),
                        clientSecret = clientSecret ?: throw IllegalStateException("Missing clientSecret for video service client. Please configure video-service.client-secret")
                )
        )
    }
}